packages/ember-data/lib/system/relationships/belongs-to.js

```diff
@@ -93,6 +93,14 @@ function belongsTo(modelName, options) {
 
   opts = opts || {};
 
+  if (opts.hasOwnProperty('serialize')) {
+    Ember.warn('You provided a serialize option, this belongs in the serializer. See DS.Serializer and it\'s implementations http://emberjs.com/api/data/classes/DS.Serializer.html');
```

Comment:
If you move the warning into the getter, you should be able to do this. Perf isn't as important as anything inside Ember.warn/assert etc. gets stripped in production